torch/_inductor/codegen/triton_foreach.py

```diff
@@ -0,0 +1,295 @@
+                    return store_cache[list_name]
+                return self.load_list(list_name, layouts)
+
+            @staticmethod
+            def store(list_name, layouts, value):
+                self.cse.store_cache[list_name] = value
+                return self.store_list(list_name, layouts, value)
+
+            @staticmethod
+            def reduction(name, dtype, src_dtype, reduction_type, index, value):
+                raise NotImplementedError()
+
+        parent_handler = self.overrides(V.get_ops_handler())
+        self.exit_stack.enter_context(V.set_ops_handler(CSEProxy()))
+        self.exit_stack.enter_context(V.set_kernel_handler(self))
+        return self
+
+    @staticmethod
+    def _compute_num_blocks(tensor_elem_counts, block_size):
+        num_blocks = 0
+        for count in tensor_elem_counts:
+            num_blocks += ceildiv(count, block_size)
+
+        return num_blocks
+
+    def _gen_tile_ptrs(self, code):
+        block_count = 0
+        for index, num_elems in enumerate(self.tensor_elem_counts):
+            num_blocks = ceildiv(num_elems, self.block_size)
+            upper_bound_pid = block_count + num_blocks
+            lower_bound_pid = block_count
+            last_block_elem_count = self.block_size - (
+                num_blocks * self.block_size - num_elems
+            )
+
+            if block_count == 0:
+                cond = "if"
+                # initialize tile ptrs
+                code.splice("mask = tl.arange(0, BLOCK_SIZE) < BLOCK_SIZE\n")
+                for list_tracker in self.lists.values():
+                    code.splice(
+                        f"{list_tracker.var}_tile_ptrs = {list_tracker.arg_names[index]} + tl.arange(0, BLOCK_SIZE)"
+                    )
+            else:
+                cond = "elif"
+
+            code.splice(f"{cond} pid >= {lower_bound_pid} and pid < {upper_bound_pid}:")
+            with code.indent():
+                code.splice(f"xoffset = (pid - {lower_bound_pid}) * BLOCK_SIZE")
+                code.splice("xindex = xoffset + tl.arange(0, BLOCK_SIZE)")
+                for list_tracker in self.lists.values():
+                    list_tracker.codegen_tile_ptrs(code, index)
+
+                code.splice(f"if pid == {upper_bound_pid - 1}:")
+                with code.indent():
+                    code.splice(
+                        f"mask = tl.arange(0, BLOCK_SIZE) < {last_block_elem_count}"
+                    )
+
+            block_count += num_blocks
+
+    def jit_line(self):
+        _, _, signature = self.args.python_argdefs()
+        triton_meta = {
+            "signature": dict(enumerate(map(signature_of, signature))),
+            "device": V.graph.scheduler.current_device.index,
+            "constants": {},
+        }
+        triton_meta["configs"] = [config_of(signature)]
+        return (
+            f"@template(num_stages=1, num_warps={self.num_warps}, meta={triton_meta!r})\n"
```

Comment:
should this live in triton_heuristics.py, similar to other configs?

Reply:
not sure I understand, I reused the template config which is already in heuristics?